qa/tasks/cephfs/fuse_mount.py

```diff
@@ -424,20 +425,27 @@ def find_socket(client_name):
         for f in files:
                 pid = re.match(".*\.(\d+)\.asok$", f).group(1)
                 if os.path.exists("/proc/{{0}}".format(pid)):
-                        return f
+                    with open("/proc/{{0}}/cmdline".format(pid), 'r') as proc_f:
+                        if mountpoint in proc_f.read():
+                            return f
```

Comment:
This still does not work for me and I am testing it on CentOS 7:

```

2020-05-11 10:32:57,835.835 INFO:__main__:Ran 1 test in 109.519s
2020-05-11 10:32:57,835.835 INFO:__main__:
2020-05-11 10:32:57,835.835 INFO:__main__:FAILED (errors=1)
2020-05-11 10:32:57,835.835 INFO:__main__:
2020-05-11 10:32:57,835.835 INFO:__main__:======================================================================
2020-05-11 10:32:57,835.835 INFO:__main__:ERROR: test_readonly_authorization (tasks.cephfs.test_volume_client.TestVolumeClient)
2020-05-11 10:32:57,835.835 INFO:__main__:----------------------------------------------------------------------
2020-05-11 10:32:57,835.835 INFO:__main__:Traceback (most recent call last):
2020-05-11 10:32:57,836.836 INFO:__main__:  File "/data/test/ceph/qa/tasks/cephfs/test_volume_client.py", line 19, in setUp
2020-05-11 10:32:57,836.836 INFO:__main__:    CephFSTestCase.setUp(self)
2020-05-11 10:32:57,836.836 INFO:__main__:  File "/data/test/ceph/qa/tasks/cephfs/cephfs_test_case.py", line 145, in setUp
2020-05-11 10:32:57,836.836 INFO:__main__:    self.mounts[i].mount_wait()
2020-05-11 10:32:57,836.836 INFO:__main__:  File "/data/test/ceph/qa/tasks/cephfs/mount.py", line 420, in mount_wait
2020-05-11 10:32:57,836.836 INFO:__main__:    mount_options=mount_options)
2020-05-11 10:32:57,836.836 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 736, in mount
2020-05-11 10:32:57,836.836 INFO:__main__:    self._set_fuse_daemon_pid()
2020-05-11 10:32:57,836.836 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 786, in _set_fuse_daemon_pid
2020-05-11 10:32:57,836.836 INFO:__main__:    raise RuntimeError('Failed to get the PID of ceph-fuse process '
2020-05-11 10:32:57,836.836 INFO:__main__:RuntimeError: Failed to get the PID of ceph-fuse process not containing sudo in args.
2020-05-11 10:32:57,836.836 INFO:__main__:
```

```
root     13483  0.0  0.0 239152  7820 pts/0    S    10:29   0:00 /usr/bin/sudo -E LD_LIBRARY_PATH=/opt/rh/devtoolset-8/root/usr/lib64:/opt/rh/devtoolset-8/root/usr/lib:/opt/rh/devtoolset-8/root/usr/lib64/dyninst:/opt/rh/devtoolset-8/root/usr/lib/dyninst:/opt/rh/devtoolset-8/root/usr/lib64:/opt/rh/devtoolset-8/root/usr/lib:/data/test/ceph/build/lib/ PATH=/opt/rh/devtoolset-8/root/usr/bin:/usr/lib64/ccache:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin scl enable devtoolset-8  'nsenter' '--net=/var/run/netns/ceph-ns--tmp-tmpe_63x2wp-mnt.0' './bin/ceph-fuse' '-f' '--name' 'client.0' '/tmp/tmpe_63x2wp/mnt.0'
root     13485  0.0  0.0      0     0 ?        S    10:29   0:00 [scsi_eh_3]
root     13487  0.0  0.0      0     0 ?        I<   10:29   0:00 [scsi_tmf_3]
root     13488  0.0  0.0      0     0 ?        I<   10:29   0:00 [iscsi_q_3]
root     13490  0.0  0.0      0     0 ?        I<   10:29   0:00 [scsi_wq_3]
root     13503  0.0  0.0   4372  1484 pts/0    S    10:29   0:00 scl enable devtoolset-8  'nsenter' '--net=/var/run/netns/ceph-ns--tmp-tmpe_63x2wp-mnt.0' './bin/ceph-fuse' '-f' '--name' 'client.0' '/tmp/tmpe_63x2wp/mnt.0'
root     13505  0.0  0.0 113204  2608 pts/0    S    10:29   0:00 /bin/bash /var/tmp/sclbhJQwq
root     13506  0.0  0.0      0     0 ?        S    10:29   0:00 [scsi_eh_4]
root     13507  0.0  0.0      0     0 ?        I<   10:29   0:00 [scsi_tmf_4]
root     13508  0.0  0.0      0     0 ?        I<   10:29   0:00 [iscsi_q_4]
root     13511  0.0  0.0      0     0 ?        I<   10:29   0:00 [scsi_wq_4]
root     13520  0.2  0.2 1886960 38884 pts/0   Sl   10:29   0:00 ./bin/ceph-fuse -f --name client.0 /tmp/tmpe_63x2wp/mnt.0
```
Maybe we should get rid of the `sudo ...` and `scl enable ...`. 
What's yours ? Are you using the Fedora ? Has there any `scl enable ...` process running ?

Reply:
Are you using the Fedora ? Has there any scl enable ... process running ?

Yep, I am using Fedora. scl's not used on/needed for it

Reply:
@lxbsz Are you still hitting this issue?

Reply:
Yeah, please see https://github.com/ceph/ceph/pull/34838#discussion_r423393537, the self.run_python() could work for me on CentOS 7 node.

Reply:
Locally without the run_python(), it will always return None instead, so strange.

Reply:
That's really odd.

Reply:
> Yeah, please see #34838 (comment), the self.run_python() could work for me on CentOS 7 node.

On it.